src/uiprotect/data/bootstrap.py

```diff
@@ -540,13 +594,16 @@ def _make_ws_packet_message(
             return None
 
         action_action: str = action["action"]
-        if action_action == "remove":
-            return self._process_remove_packet(model_type, action)
-
-        if not data and not is_ping_back:
-            return None
 
         try:
+            if model_type in {ModelType.KEYRING, ModelType.ULP_USER}:
+                return self._process_ws_keyring_or_ulp_user_message(
+                    action, data, model_type
+                )
+            if action_action == "remove":
+                return self._process_remove_packet(model_type, action)
```

Comment:
If this is existing missing coverage, please add a github review comment

Reply:
it's existing missing coverage